<a href="https://colab.research.google.com/github/Almehedi06/Statistics-with-Python-/blob/main/ANOVA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from pandas import read_csv
from matplotlib import pyplot
# To remove the scientific notation from numpy arrays
np.set_printoptions(suppress=True)

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# One-way 

https://www.statology.org/one-way-anova-python/

A researcher recruits 30 students to participate in a study. The students are randomly assigned to use one of three studying techniques for the next three weeks to prepare for an exam. At the end of the three weeks, all of the students take the same test. 

Use the following steps to perform a one-way ANOVA to determine if the average scores are the same across all three groups.

In [ ]:
#enter exam scores for each group
group1 = [85, 86, 88, 75, 78, 94, 98, 79, 71, 80]
group2 = [91, 92, 93, 85, 87, 84, 82, 88, 95, 96]
group3 = [79, 78, 88, 94, 92, 85, 83, 85, 82, 81]

we’ll use the f_oneway() function from the SciPy library to perform the one-way ANOVA:

In [ ]:
from scipy.stats import f_oneway

#perform one-way ANOVA
f_oneway(group1, group2, group3)

F_onewayResult(statistic=2.3575322551335636, pvalue=0.11384795345837218)

Interpret the results.

A one-way ANOVA uses the following null and alternative hypotheses:

    H0 (null hypothesis): μ1 = μ2 = μ3 = … = μk (all the population means are equal)
    H1 (null hypothesis): at least one population mean is different from the rest

The F test statistic is 2.3575 and the corresponding p-value is 0.1138. Since the p-value is not less than .05, we fail to reject the null hypothesis.

This means we do not have sufficient evidence to say that there is a difference in exam scores among the three studying techniques.

# Two-way

https://www.statology.org/two-way-anova-python/

A botanist wants to know whether or not plant growth is influenced by sunlight exposure and watering frequency. She plants 30 seeds and lets them grow for two months under different conditions for sunlight exposure and watering frequency. After two months, she records the height of each plant, in inches.

Use the following steps to perform a two-way ANOVA to determine if watering frequency and sunlight exposure have a significant effect on plant growth, and to determine if there is any interaction effect between watering frequency and sunlight exposure.

Water: how frequently each plant was watered: daily or weekly

Sun: how much sunlight exposure each plant received: low, medium, or high

Height: the height of each plant (in inches) after two months


In [ ]:
import numpy as np
import pandas as pd

#create data
df = pd.DataFrame({'water': np.repeat(['daily', 'weekly'], 15),
                   'sun': np.tile(np.repeat(['low', 'med', 'high'], 5), 2),
                   'height': [6, 6, 6, 5, 6, 5, 5, 6, 4, 5,
                              6, 6, 7, 8, 7, 3, 4, 4, 4, 5,
                              4, 4, 4, 4, 4, 5, 6, 6, 7, 8]})

In [ ]:
df[:10]

water  sun  height
0  daily  low       6
1  daily  low       6
2  daily  low       6
3  daily  low       5
4  daily  low       6
5  daily  med       5
6  daily  med       5
7  daily  med       6
8  daily  med       4
9  daily  med       5

In [ ]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

#perform two-way ANOVA
model = ols('height ~ C(water) + C(sun) + C(water):C(sun)', data=df).fit()
sm.stats.anova_lm(model, typ=2)

sum_sq    df        F    PR(>F)
C(water)          8.533333   1.0  16.0000  0.000527
C(sun)           24.866667   2.0  23.3125  0.000002
C(water):C(sun)   2.466667   2.0   2.3125  0.120667
Residual         12.800000  24.0      NaN       NaN

We can see the following p-values for each of the factors in the table:

water: p-value = .000527

sun: p-value = .0000002

water*sun: p-value = .120667